File: Entregable4.2/src/main/java/pojo/MostSoldProduct.java
package pojo;

public class MostSoldProduct {
	
	private String name;
	private double amount;
	private Long quantity;
	private double pricePerUnit;
	
	public MostSoldProduct() {}
	
	public MostSoldProduct(String name, double amount, Long quantity, double pricePerUnit) {
		this.name = name;
		this.amount = amount;
		this.quantity = quantity;
		this.pricePerUnit = pricePerUnit;
	}

	public String getName() {
		return name;
	}

	public double getAmount() {
		return amount;
	}

	public Long getQuantity() {
		return quantity;
	}
	
	public double getPricePerUnit() {
		return pricePerUnit;
	}
}

File: Entregable5/src/main/resources/static/js/mostSoldProduct.js
document.addEventListener("DOMContentLoaded", () => {

    const getMostSoldProduct = async() => {
        const productBox = document.querySelector("#product");

        try {
            productBox.innerHTML = `
            <div class="spinner-border text-light" role="status"></div>
            `
            const url = "/tickets/most-sold-product"
            const response = await fetch(url);
            const product = await response.json();
console.log(response, product);
            productBox.innerHTML = '';

            productBox.innerHTML += `
            <div class="card text-white bg-dark mb-3">
                <div class="card-body">
                    <h5 class="card-title">Nombre: ${product.name}</h5>
                    <h5 class="card-title">Monto total: $${product.amount}</h5>
                    <h5 class="card-title">Cantidad: ${product.quantity}</h5>
                    <h5 class="card-title">Precio por unidad: $${product.pricePerUnit}</h5>
                </div>
            </div>
            `   
        } catch (response) {
            console.log("Algo falló.");
        }
    }

    (async () => {
        try {
            await getMostSoldProduct();
        } catch (e) {
        }
    })();
})